test/url-params.test.ts

```diff
@@ -262,9 +265,48 @@ test('set default client', () => {
 	urlParams.consumeUrlParameters('optOutAnalytics=false');
 	expect(urlParams.isOptOutRequested()).toBeFalsy();
 
+	const passedDeviceId = urlParams.getPassedDeviceId();
 	const client = clientMock();
-	const initialDeviceId = client.deviceId();
+
+	if (passedDeviceId) {
+		client.setDeviceId(passedDeviceId);
+	} else {
+		client.deviceId();
+	}
+	urlParams.setClient(client);
+	const allDeviceIds = urlParams.allDeviceIds();
+	expect(client.setDeviceIdParams).toStrictEqual(null);
+	expect(allDeviceIds).toEqual(['test_device_id']);
+	expect(client.knownSessionId).toStrictEqual(123);
+});
+
+test('set default client with passed sessionId', () => {
+	const urlParams = new AnalyticsUrlParams();
+	urlParams.consumeUrlParameters('s_id=999&optOutAnalytics=false');
+	expect(urlParams.isOptOutRequested()).toBeFalsy();
+	const passedSessonId = urlParams.getSessionId();
+
+	const client = clientMock();
+	urlParams.setClient(client);
+	expect(client.setDeviceIdParams).toStrictEqual(null);
+	expect(client.knownSessionId).toStrictEqual(passedSessonId);
+});
```

Comment:
Shall we add one more check here to verify that `AnalyticsUrlParams.client` is equal to the `clientMock()`?

If we had that check, the test would had failed because of my earlier comment on the `return` part after sessionId setting :nerd_face:

Reply:
@gelbal  How can we do that? `AnalyticsUrlParams.client` doesnt exists per se and we dont have a method yet to retrieve the client. Even more, the variable names on each are also different, like for example `knownSessionId`

Reply:
Ah ok. Yeah I was asking like, if it is worth implementing a getClient method just for that test. And also, I dont understand that ClientMock really well so I dont know if I should expect it to be equal to urlParams Client since they have different variables names and stuff. But if you say it should, then I'll go on and implement getClient and see what happens. let me try and push it so you can see.